docs/concepts/ClusterResourcePlacement/README.md

```diff
@@ -0,0 +1,222 @@
+In addition, each time I update my workloads, the updates take place with zero downtime by rolling out to these three clusters incrementally.
+
+## Placement Workflow
+
+![](placement-concept-overview.jpg)
+
+The placement workflow will be divided into several stages:
+* Creating `ClusterSchedulingPolicySnapshot` and `ClusterResourceSnapshot` snapshots: clusterResourcePlacement controller 
+captures the selected resources and placement policy specified in the `ClusterResourcePlacement`.
+* Creating `ClusterResourceBinding`: multi-cluster scheduler creates the `clusterResourceBinding` based on the scheduling 
+decisions for each target cluster and rollout controller binds selected resources according to the rollout strategy.
+* Creating `Work` on the member cluster reserved namespace of the hub cluster: work generator creates the work on the 
+corresponding member cluster namespace. Each work contains the manifest workload to be deployed on the member clusters.
+* Applying `Work` on the member clusters: apply work controller applies the manifest workload on the member clusters.
+
+## Resource Selection
+
+Resource selectors identify cluster-scoped objects to include based on standard Kubernetes identifiers - namely, the `group`, 
+`kind`, `version`, and `name` of the object. Namespace-scoped objects are included automatically when the namespace they
+are part of is selected. The example `ClusterResourcePlacement` above would include the `test-deployment` namespace and 
+any objects that were created in that namespace.
+
+The clusterResourcePlacement controller creates the `ClusterResourceSnapshot` to store a snapshot of selected resources
+selected by the placement. The `ClusterResourceSnapshot` spec is immutable. Each time when the selected resources are updated,
+the clusterResourcePlacement controller will detect the resource changes and create a new `ClusterResourceSnapshot`. It implies
+that resources can change independently of any modifications to the `ClusterResourceSnapshot`. In other words, resource
+changes can occur without directly affecting the `ClusterResourceSnapshot` itself.
+
+The total amount of selected resources may exceed the 1MB limit for a single Kubernetes object. As a result, the controller 
+may produce more than one `ClusterResourceSnapshot`s for all the selected resources.
+
+## Placement Policy
+
+`ClusterResourcePlacement` supports three types of policy as mentioned above. `ClusterSchedulingPolicySnapshot` will be
+generated whenever policy changes are made to the `ClusterResourcePlacement` that require a new scheduling. Similar to
+`ClusterResourceSnapshot`, its spec is immutable.
+
+Following the Kubernetes' original scheduler framework, the multi-cluster scheduling selects a cluster for the placement 
+in a 2-step operation:
+1. Filtering
+2. Scoring
+
+The _filtering_ step finds the set of clusters where it's feasible to schedule the placement, for example, whether the cluster

```

Comment:
I think an example/diagram of this would helpful

Reply:
Added a diagram. More detailed information will be added into the scheduler readme.